pkg/authorization/delegating_authorizer.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package authorization
+
+import (
+	"context"
+
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+)
+
+// NewConditionalDelegatingAuthorizer returns an authorizer that takes an authorizer
+// and delegates if the authorization decision is Allowed,
+// then returning the delegate's decision, reason, and error.
+//
+// Else, the delegate is not invoked at all
+// and the authorizer's decision (NoDecision, Denied), reason, and error is returned.
+func NewConditionalDelegatingAuthorizer(primary, delegate authorizer.Authorizer) authorizer.Authorizer {
```

Comment:
not sure I like this construct. It's hard to understand what's going on. Would rather drop the annotations (and logging) for the delegation case.

Reply:
yes, it's ugly, let's remove it. I'll:
- remove the delegation authorizer
- reintroduce delegation to authorizers
- drop audit logs in case of Allow decisions

Reply:
done